rvip/include/rvip/cuboid_generation.hpp

```diff
@@ -0,0 +1,394 @@
+        pcl::PointCloud<pcl::PointXYZ>::Ptr objplaneCloud (new pcl::PointCloud<pcl::PointXYZ>);
+        *objplaneCloud = *yoloPointCloud;
+
+        getObjectTopPlane(objplaneCloud,center,std_dev);
+
+        if (objplaneCloud->points.size() == 0) {
+                ROS_WARN_STREAM("objplaneCloud size is 0 ");
+                return temp;
+        }
+
+        //Determining the centroid of the object plane
+        pcl::CentroidPoint<pcl::PointXYZ> plane_centroid;
+        pcl::PointXYZ plane_pos;
+        for (int i = 0; i < objplaneCloud->size(); i++) {
+                plane_centroid.add (objplaneCloud->at(i));
+        }
+        plane_centroid.get(plane_pos);
+
+        // Derive the length and width of the object plane.
+        std::vector<cv::Point2f> pts;
+        cv::Point2f curr_pt(0,0);
+        for (int i = 0; i < objplaneCloud->points.size (); i++) {
+                curr_pt.x = objplaneCloud->points[i].x;
+                curr_pt.y = objplaneCloud->points[i].y;
+                pts.push_back(curr_pt);
+        }
+
+        cv::RotatedRect minRect;
+        try{
+                minRect = minAreaRect(pts);
+        }
+        catch (cv::Exception& e) {
+                ROS_WARN_STREAM("minAreaRect func failed.");
+                return temp;
+        }
+
+        object.obj_width = minRect.size.width;
+        object.obj_length =  minRect.size.height;
+        object.obj_height = abs(center.z - plane_pos.z);
+
+        // DEBUG - Check the length, width and height of object detected.
+        // ROS_INFO_STREAM("[In cuboid_generation.hpp]");
+        // ROS_INFO_STREAM("[width = " << minRect.size.width << " ,length = " << minRect.size.height << " ,height =  " << abs(center.z - plane_pos.z) << "]" );
+
+        createCuboid(object, minRect.size.height, minRect.size.width, abs(center.z - plane_pos.z));
+
+        // Conducting manual passthrough filter on the yolo-box filtered pcl
+        // Passthrough, VoxelGrid, Planar Segmentation
+        pcl::PointCloud<pcl::PointXYZ>::Ptr alignCloud (new pcl::PointCloud<pcl::PointXYZ>);
+        *alignCloud = *yoloPointCloud;
+        // Based on the assumption that two planes of the object is detected by a depth sensor positioned at a 45 degrees angle.
+
+        getAlignPointCloud(alignCloud,center,std_dev);
+
+        // Update global variables of object center later used for pose alignment.
+        object.obj_x = plane_pos.x;
+        object.obj_y = plane_pos.y;
+        object.obj_z = plane_pos.z + abs(center.z - plane_pos.z)/2;
+
+        sensor_msgs::PointCloud2 outputCloud;
+        pcl::toROSMsg(*alignCloud, outputCloud);
+
+        // DEBUG - Timestamp END - Print out time taken for pose alignment.
+        // end_ = ros::WallTime::now();
+        // double execution_time = (end_ - start_).toNSec() * 1e-6;
+        // ROS_INFO("Execution time for [Cuboid Generation] (ms):  %f                  \r", execution_time);
+        // fflush(stdout);
+
+        return outputCloud;
+}
```

Comment:
This header does not appear to have any include guards.

Reply:
Addressed in [this commit](https://github.com/ros-industrial/rvip/pull/1/commits/d8d6f424b0cd678b617b8c1e1b15d1de4f998824).